src/generic_test.rs

```diff
@@ -101,31 +107,34 @@ impl PeerList<Id, Error> for TestPeerList<Id> {
 
         Ok(())
     }
-    // Loads peers in from a json file. Not relevant to this test.
+
+    /// Loads peers in from a json file. Not relevant to this test.
     fn get_peers_from_file(&mut self, _json_peer_path: String) -> std::result::Result<(), Error> {
         // Stub not used in tests to satisfy PeerList trait
         Ok(())
     }
-    // Allows iteration over the peer list.
+
+    /// Allows iteration over the peer list.
     fn iter(&self) -> Iter<'_, Self::P> {
         self.peers.iter()
     }
-    // Allows a mutable iteration over the peer list.
+
+    /// Allows a mutable iteration over the peer list.
     fn iter_mut(&mut self) -> IterMut<'_, Self::P> {
         self.peers.iter_mut()
     }
 }
 
-/*
-    The function used to actually test the Transport. It takes in a Transport Configuration and a
-    Transport trait implementor.
+/**
+    The function used to actually test a `Transport`. It takes in a `TransportConfiguration` and a
+    `Transport` trait implementor.
 
-    THis method simply takes in a list of peers, instantiates them, and tests whether they can
-    send/receive data to one another.
-*/
+    This method simply takes in a list of peers, instantiates them, and tests whether they can
+    send/receive data to/from one another.
+**/
 pub fn common_test<
-    //    C: TransportConfiguration<Data>,
-    T: Transport<Id, Data, Error, TestPeerList<Id>>,
+    C: TransportConfiguration<Data>,
```

Comment:
That was removed in the latest master branch, why it is reverted back here?

Reply:
Because that change in `master` is a regression and I did not agree with it. It is better to pass a configuration object to `Transport::new`.